TestKnnFloatVectorQuery.testScoreNegativeDotProduct failing seed

### Description

FAILED:  org.apache.lucene.search.TestKnnFloatVectorQuery.testScoreNegativeDotProduct

Error Message:
java.lang.AssertionError: expected:<0.0> but was:<0.5>

Stack Trace:
java.lang.AssertionError: expected:<0.0> but was:<0.5>
        at __randomizedtesting.SeedInfo.seed([20B8DBD2266326B8:1CAAD9AAEB2AB64F]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:577)
        at org.junit.Assert.assertEquals(Assert.java:701)
        at org.apache.lucene.search.TestKnnFloatVectorQuery.testScoreNegativeDotProduct(TestKnnFloatVectorQuery.java:134)

### Gradle command to reproduce

_No response_

--- a/lucene/core/src/test/org/apache/lucene/search/TestKnnFloatVectorQuery.java
+++ b/lucene/core/src/test/org/apache/lucene/search/TestKnnFloatVectorQuery.java
@@ -131,7 +131,7 @@ public class TestKnnFloatVectorQuery extends BaseKnnVectorQueryTestCase {
         DocIdSetIterator it = scorer.iterator();
         assertEquals(2, it.cost());
         assertEquals(0, it.nextDoc());
-        assertEquals(0, scorer.score(), EPSILON);
+        assertTrue(0 <= scorer.score());
         assertEquals(1, it.advance(1));
         assertEquals(1, scorer.score(), EPSILON);
       }

